lib/rules/key-spacing.js

```diff
@@ -166,12 +189,14 @@ module.exports = function(context) {
      * @param {string} side Side being verified - either "key" or "value".
      * @param {string} whitespace Actual whitespace string.
      * @param {int} expected Expected whitespace length.
+     * @param {boolean} singleLine True if it's single line check
      * @returns {void}
      */
-    function report(property, side, whitespace, expected) {
+    function report(property, side, whitespace, expected, singleLine) {
```

Comment:
Instead of passing in a boolean `singleLine` flag here and in `verifySpacing`, can you instead pass in the options directly so the functions don't have to retrieve them themselves? I think passing in `mode` here makes sense from a design standpoint - the last three in order would be actual spacing, expected spacing, and the mode for the comparison between the two.